eng/scripts/TypeSpec-Requirement.ps1

```diff
@@ -98,27 +98,51 @@ else {
 
     try {
```

Comment:
I think it would be great if you could add somewhere in the header comment of this script that it assumes the parsed directory structure adheres to:

- https://github.com/Azure/azure-rest-api-specs/blob/main/README.md#directory-structure

with some possible differences / violations of rules.

I believe this would help the reader a lot to figure out what this code is trying to achieve.

Reply:
Example file paths are documented here:

https://github.com/Azure/azure-rest-api-specs/pull/28444/files#diff-3c37fefe028ad9bd154b76ea3371f8db2d7a7609dafa323e92d8461880e808b2R82-R84

Do you just want to me to add a link to `README.md#directory-structure` above these examples?

Reply:
Yeah, that would be great. But I think we probably want to clarify this is "aspirational"; that many exceptions to the directory structure exist as of today in the repo, and this script also tried to handle some of them.